bears/c_languages/ArtisticStyleBear.py

```diff
@@ -0,0 +1,873 @@
+    DistributionRequirement)
+
+
+@linter(executable='astyle',
+        output_format='corrected',
+        use_stdin=True)
+class ArtisticStyleBear:
+    """
+    Artistic Style is a source code indenter, formatter,
+    and beautifier for the C, C++, C++/CLI, Objective-C,
+    C# and Java programming languages.
+    """
+
+    LANGUAGES = {'C', 'C++', 'Objective-C', 'C#', 'Java'}
+    REQUIREMENTS = {
+        DistributionRequirement(
+            apt_get='astyle',
+            dnf='astyle'
+        )
+    }
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_FIX = {'Formatting'}
+    SEE_MORE = 'http://astyle.sourceforge.net/astyle.html'
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         bracket_style: str='',
+                         use_spaces: bool=None,
+                         indent_size: int=4,
+                         attach_braces_to_namespace: bool=True,
+                         attach_braces_to_class: bool=True,
+                         attach_braces_to_inline: bool=True,
+                         attach_braces_to_extern: bool=False,
+                         allow_indent_classes: bool=True,
+                         allow_indent_modifiers: bool=True,
+                         allow_indent_switches: bool=True,
+                         allow_indent_cases: bool=True,
+                         allow_indent_namespaces: bool=False,
+                         allow_indent_labels: bool=True,
+                         allow_indent_preproc_block: bool=True,
+                         allow_indent_preproc_definition: bool=True,
+                         allow_indent_preproc_conditionals: bool=True,
+                         allow_indent_column_one_comments: bool=True,
+                         allow_pad_header_blocks: bool=True,
+                         allow_pad_operators: bool=True,
+                         allow_pad_parenthesis: bool=False,
+                         allow_pad_parenthesis_out: bool=False,
+                         allow_pad_parenthesis_in: bool=False,
+                         delete_empty_lines_in_func: bool=False,
+                         break_closing_braces: bool=False,
+                         break_elseifs: bool=False,
+                         break_one_line_headers: bool=False,
+                         add_braces_to_one_line_conditionals: bool=False,
+                         remove_braces_from_one_line_conditionals: bool=False,
+                         remove_comment_prefix: bool=True):
+        """
+        :param bracket_style:
+            Defines the brace style to use.
+            Possible values are ``allman, java, kr, stroustrup, whitesmith,
+            vtk, banner, gnu, linux, horstmann, google, mozilla, pico and
+            lisp.``
+            For example: Allman style uses braces that are broken from the
```

Comment:
See https://github.com/coala/coala-bears/pull/1882#discussion_r128701045. Referring to the actual keyword/parameter may cause problems in some cases like Kernighan & Ritchie. The actual keyword is referred to as `kr` so replacing `Kernighan & Ritchie` with `kr` may make it difficult for users to understand the style name.